### PR TITLE
Add nodeSelector to storageClass

### PR DESCRIPTION
**Pull request by msoGTA**
This PR is link to the following issue [https://github.com/longhorn/longhorn/issues/4574]
 add possibility to use nodeSelector on the storageClass #4574

### CODE REVIEW

**Comment by mantissahz** [comment]
@msoGTA appreciate your contribution,
Check the DCO failed, `There are 2 commits incorrectly signed off`.
Please use `git commit -s` to sign-off in your commits. Thanks.

File: chart/values.yaml [hunk 1]
```diff
@@ -85,6 +85,10 @@ persistence:
     dataSourceType: ~
     dataSourceParameters: ~
     expectedChecksum: ~
+   defaultNodeSelector:
+    enable: false # disable by default
+    name: longhorn # longhorn is the value of the label and can be named differnetly
```
Comment (innobead) [inline chart/values.yaml:90]: Suggest renaming `name` to `selector` and the default value should be `[]`, because it supports type list instead.

**Review by innobead — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Please also update questions.yaml which is used for Rancher chart interactive installation.

File: chart/values.yaml (hunk 1 above)
Comment (innobead) [reply chart/values.yaml:90]: cc @c3y1huang we should make the same pattern for `recurringJobSelector`. You can create a ticket for future refactoring. 

```yaml
  recurringJobSelector:
    enable: false
    jobList: []
```

Comment (msoGTA) [reply chart/values.yaml]: Hi innobead, thanks for your review. I just pushed a new commit with your suggestions. Regards.

Comment (innobead) [reply chart/values.yaml]: Thanks, please consolidate your commits to one, and update questions.yaml (you can reference previous commit history to know how to update that from other settings) as I mentioned above. Also, remember to sign off the commit.

Comment (msoGTA) [reply chart/values.yaml]: Hi, I made the requested changes. 
Regards

**Review by innobead — APPROVED** [review]
LGTM

But just notice some naming of settings in persistence seems are inconsistent, so we probably need to refactor them later. (minor)

**Comment by innobead** [comment]
@weizhe0422 Please help handle this contribution later at https://github.com/longhorn/longhorn/issues/4574. Thanks. 

@msoGTA thanks for the contribution!

**Comment by msoGTA** [comment]
You are welcome :) @innobead

**Pull request merged**